### Phân tích dữ liệu

In [ ]:
import pandas as pd 
import numpy as np 

# 1.Đọc dữ liệu

In [ ]:
%cd "C:\Users\sv\Documents\A41131"

[Errno 2] No such file or directory: 'C:\\Users\\sv\\Documents\\A41131'
/content


In [ ]:
data = pd.read_csv('Life Expectancy Data.csv')

# 2.Tìm hiểu sơ bộ về dữ liệu

## 2.1 Kiểm tra thông tin cơ bản

In [ ]:
data.head()

Country  Year      Status  Life expectancy   Adult Mortality  \
0  Afghanistan  2015  Developing              65.0            263.0   
1  Afghanistan  2014  Developing              59.9            271.0   
2  Afghanistan  2013  Developing              59.9            268.0   
3  Afghanistan  2012  Developing              59.5            272.0   
4  Afghanistan  2011  Developing              59.2            275.0   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   ...  \
0             62     0.01               71.279624         65.0      1154  ...   
1             64     0.01               73.523582         62.0       492  ...   
2             66     0.01               73.219243         64.0       430  ...   
3             69     0.01               78.184215         67.0      2787  ...   
4             71     0.01                7.097109         68.0      3013  ...   

   Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0    6.0               8.16         65.0        0.1  584.259210  33736494.0   
1   58.0               8.18         62.0        0.1  612.696514    327582.0   
2   62.0               8.13         64.0        0.1  631.744976  31731688.0   
3   67.0               8.52         67.0        0.1  669.959000   3696958.0   
4   68.0               7.87         68.0        0.1   63.537231   2978599.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  

[5 rows x 22 columns]

Một số thông tin bảng dữ liệu:
- Country: Tên quốc gia
- Year: Năm
- Status: Tình trạng của quốc gia nhất định (Đang phát triển hoặc Đã phát triển)
- Life expectancy: tính bằng năm
- Adult Mortality: Tỷ lệ tử vong ở người trưởng thành của cả hai giới (xác suất tử vong từ 15 đến 60 tuổi trên 1000 dân số)
- Infant deaths: Số trẻ sơ sinh tử vong trên 1000 dân số
- Alcoholv: Rượu, được ghi nhận mức tiêu thụ bình quân đầu người (15+) (tính bằng lít rượu nguyên chất)
- Percentage expenditure: Chi tiêu cho y tế tính theo phần trăm Tổng sản phẩm quốc nội trên đầu người (%)
- Hepatitis B: Tỷ lệ tiêm chủng viêm gan B (HepB) ở trẻ 1 tuổi (%)
- Measles: Sởi - số trường hợp được báo cáo trên 1000 dân số
- BMI: Chỉ số khối cơ thể trung bình của toàn bộ dân số
- Under five death- Số người chết dưới 5 tuổi trên 1000 dân số

- Polio  - Tỷ lệ tiêm chủng bại liệt (Pol3) ở trẻ 1 tuổi (%)

- Total expenditure - Chi tiêu chung của chính phủ cho y tế tính theo tỷ lệ phần trăm của tổng chi tiêu của chính phủ (%)

- Diphtheria - Tỷ lệ tiêm chủng giải độc tố uốn ván và ho gà (DTP3) cho trẻ 1 tuổi (%)

- HIV / AID - Tử vong trên 1000 trẻ sinh sống HIV / AIDS (0-4 tuổi)

- GDP - Tổng sản phẩm quốc nội bình quân đầu người (tính bằng USD)

- Population - Dân số cả nước

- thinness 1-19 years - Tỷ lệ gầy ở trẻ em và thanh thiếu niên từ 10 đến 19 tuổi (%)

- thinness 5-9 years - Tỷ lệ gầy ốm ở trẻ em từ 5-9 tuổi (%)

- Income composition of resources - Chỉ số Phát triển Con người xét theo cơ cấu thu nhập của các nguồn lực (chỉ số từ 0 đến 1)
- Schooling - Số năm đi học (năm)


In [ ]:
data.shape 

(2938, 22)

- Dữ liêu có 2938 hàng,22 cột

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2938 entries, 0 to 2937
Data columns (total 22 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Country                          2938 non-null   object 
 1   Year                             2938 non-null   int64  
 2   Status                           2938 non-null   object 
 3   Life expectancy                  2928 non-null   float64
 4   Adult Mortality                  2928 non-null   float64
 5   infant deaths                    2938 non-null   int64  
 6   Alcohol                          2744 non-null   float64
 7   percentage expenditure           2938 non-null   float64
 8   Hepatitis B                      2385 non-null   float64
 9   Measles                          2938 non-null   int64  
 10   BMI                             2904 non-null   float64
 11  under-five deaths                2938 non-null   int64  
 12  Polio               

Một số thông tin sau khi kiểm tra:
- Cột Alcohol,Adult Mortality,Life expectancy, Hepatitis B, BMI,Polio,Total expenditure,Diphtheria,GDP, Population,thinness  1-19 years,thinness 5-9 years,Income composition of resources,Schooling xuất hiện giá trị bị thiếu
- Các cột có kiểu dữ liệu định lượng là các cột có Dtype: int64, float64
- Các cột có kiểu dữ liệu định tính là các cột có Dtype: object

In [ ]:
data.columns

Index(['Country', 'Year', 'Status', 'Life expectancy ', 'Adult Mortality',
       'infant deaths', 'Alcohol', 'percentage expenditure', 'Hepatitis B',
       'Measles ', ' BMI ', 'under-five deaths ', 'Polio', 'Total expenditure',
       'Diphtheria ', ' HIV/AIDS', 'GDP', 'Population',
       ' thinness  1-19 years', ' thinness 5-9 years',
       'Income composition of resources', 'Schooling'],
      dtype='object')

## 2.2 Kiểm tra vấn đề bất thường

### a.Tính toán số lượng missing value 

In [ ]:
data.isnull().sum()

Country                              0
Year                                 0
Status                               0
Life expectancy                     10
Adult Mortality                     10
infant deaths                        0
Alcohol                            194
percentage expenditure               0
Hepatitis B                        553
Measles                              0
 BMI                                34
under-five deaths                    0
Polio                               19
Total expenditure                  226
Diphtheria                          19
 HIV/AIDS                            0
GDP                                448
Population                         652
 thinness  1-19 years               34
 thinness 5-9 years                 34
Income composition of resources    167
Schooling                          163
dtype: int64

In [ ]:
(data.isnull().sum()/(len(data)))*100

Country                             0.000000
Year                                0.000000
Status                              0.000000
Life expectancy                     0.340368
Adult Mortality                     0.340368
infant deaths                       0.000000
Alcohol                             6.603131
percentage expenditure              0.000000
Hepatitis B                        18.822328
Measles                             0.000000
 BMI                                1.157250
under-five deaths                   0.000000
Polio                               0.646698
Total expenditure                   7.692308
Diphtheria                          0.646698
 HIV/AIDS                           0.000000
GDP                                15.248468
Population                         22.191967
 thinness  1-19 years               1.157250
 thinness 5-9 years                 1.157250
Income composition of resources     5.684139
Schooling                           5.547992
dtype: flo

Tính toán tỉ lệ bị thiếu trên tổng quan sát của mỗi cột. Có thể thấy dữ liệu:
- Cột Hepatitis B gặp vấn đề đến ~ 18.8% quan sát gặp vấn đề bị thiếu
- Cột BMI gặp vấn đề 1% quan sát gặp vấn đề bị thiếu
- Cột Total expenditure gặp vấn đề đến ~ 7.6% quan sát gặp vấn đề bị thiếu
- Cột GDP gặp vấn đề đến ~15.2% quan sát gặp vấn đề bị thiếu
- Population ~22,1% dữ liệu bị thiếu
- thinness  1-19 years và cột thinness 5-9 years ~1.15% dữ liệu bị thiếu
- Schooling ~6% bị thiêu 


### b.Kiểm tra vấn đề bị lặp

In [ ]:
data.nunique()

Country                             193
Year                                 16
Status                                2
Life expectancy                     362
Adult Mortality                     425
infant deaths                       209
Alcohol                            1076
percentage expenditure             2328
Hepatitis B                          87
Measles                             958
 BMI                                608
under-five deaths                   252
Polio                                73
Total expenditure                   818
Diphtheria                           81
 HIV/AIDS                           200
GDP                                2490
Population                         2278
 thinness  1-19 years               200
 thinness 5-9 years                 207
Income composition of resources     625
Schooling                           173
dtype: int64

In [1]:
data.duplicated(subset=['Country','Year']).sum() # lựa chọn 1 cột

NameError: ignored

In [ ]:
data.duplicated().sum()# kiểm tra liệu có hai dòng nào giống nhau hoàn toàn hay không

0